extensions/notification/NotificationQueuePopulator.js

```diff
@@ -42,6 +42,75 @@ class NotificationQueuePopulator extends QueuePopulatorExtension {
             || key === undefined);
     }
 
+    /**
+     * Get bucket attributes from log entry
+     *
+     * @param {Object} value - log entry object
+     * @return {Object|undefined} - bucket attributes if available
+     */
+    _getBucketAttributes(value) {
+        if (value && value.attributes) {
```

Comment:
should we cache this in `this._bucketAttributes` or something?(Lazy Initialization of _getBucketAttributes)